ipn/ipnlocal/local.go

```diff
@@ -267,9 +270,93 @@ func NewLocalBackend(logf logger.Logf, logid string, store ipn.StateStore, diale
 		b.logf("[unexpected] failed to wire up peer API port for engine %T", e)
 	}
 
+	for _, component := range debuggableComponents {
+		key := componentStateKey(component)
+		if v, err := store.ReadState(key); err == nil {
+			if ut, err := strconv.ParseInt(string(v), 10, 64); err == nil {
+				until := time.Unix(ut, 0)
+				if until.After(time.Now()) { // conditional to avoid log spam at start when off
+					b.SetComponentDebugLogging(component, until)
+				}
+			}
```

Comment:
so much nesting :lolsob:

Reply:
yeah but it's 5 lines so still readable

Reply:
Could add a method on `store`? e.g. `store.ReadInt(key)` and/or `store.ReadTime(key)`?

Reply:
store's an interface so no, not really

Reply:
Something like this?

```diff
diff --git ipn/ipnlocal/local.go ipn/ipnlocal/local.go
index de2c68d6..da904aea 100644
--- ipn/ipnlocal/local.go
+++ ipn/ipnlocal/local.go
@@ -272,12 +272,10 @@ func NewLocalBackend(logf logger.Logf, logid string, store ipn.StateStore, diale
 
        for _, component := range debuggableComponents {
                key := componentStateKey(component)
-               if v, err := store.ReadState(key); err == nil {
-                       if ut, err := strconv.ParseInt(string(v), 10, 64); err == nil {
-                               until := time.Unix(ut, 0)
-                               if until.After(time.Now()) { // conditional to avoid log spam at start when off
-                                       b.SetComponentDebugLogging(component, until)
-                               }
+               if ut, err := ipn.ReadStoreInt(store, key); err == nil {
+                       until := time.Unix(ut, 0)
+                       if until.After(time.Now()) { // conditional to avoid log spam at start when off
+                               b.SetComponentDebugLogging(component, until)
                        }
                }
        }
diff --git ipn/store.go ipn/store.go
index 53aa5dae..8749398d 100644
--- ipn/store.go
+++ ipn/store.go
@@ -6,6 +6,7 @@
 
 import (
        "errors"
+       "strconv"
 )
 
 // ErrStateNotExist is returned by StateStore.ReadState when the
@@ -48,3 +49,23 @@ type StateStore interface {
        // WriteState saves bs as the state associated with ID.
        WriteState(id StateKey, bs []byte) error
 }
+
+// ReadStoreInt reads an integer from a StateStore.
+func ReadStoreInt(store StateStore, id StateKey) (int64, error) {
+       v, err := store.ReadState(id)
+       if err != nil {
+               return 0, err
+       }
+
+       ii, err := strconv.ParseInt(string(v), 10, 64)
+       if err != nil {
+               return 0, err
+       }
+
+       return ii, nil
+}
+
+// PutStoreInt puts an integer into a StateStore.
+func PutStoreInt(store StateStore, id StateKey, val int) error {
+       return store.WriteState(id, []byte(strconv.Itoa(val)))
+}
```

:shrug:

Reply:
Done; PTAL